src/pynwb/image.py

```diff
@@ -79,6 +79,8 @@ def __init__(self, **kwargs):
 
         if args_to_set['external_file'] is not None and args_to_set['starting_frame'] is None:
             args_to_set['starting_frame'] = [0] if len(args_to_set['external_file']) == 1 else None
+        if args_to_set['external_file'] is not None and args_to_set['format'] is None:
+            args_to_set['format'] = 'external'
```

Comment:
I'm not sure that we want a default value this case.

Reply:
As long as ``data`` is ``None`` and ``external_file`` is ``not None``, this behavior should be Ok. However, I agree that the more consistent behavior would be to just raise a ValueError here. However, since this changes behavior of the API, what I would suggest for now is to just add a warning that indicates: 1) that the value for format has been changed, and b) that this behavior is deprecated and will be changed in the next released. We can then change the behavior to a ValueError in the next main release after that.